pyhf/optimize/opt_tflow.py

```diff
@@ -11,18 +19,28 @@ def unconstrained_bestfit(self, objective, data, pdf, init_pars, par_bounds):
 
         pars      = self.tb.concatenate(parlist)
         objective = objective(pars,data,pdf)
-        hessian   = tf.hessians(objective, pars)[0]
+        hessian   = tf.hessians(objective, pars)[0]+1e-10
         gradient  = tf.gradients(objective, pars)[0]
         invhess   = tf.linalg.inv(hessian)
         update    = tf.transpose(tf.matmul(invhess, tf.transpose(tf.stack([gradient]))))[0]
 
         #run newton's method
         best_fit = init_pars
-        for i in range(1000):
-            up = self.tb.session.run(update, feed_dict={pars: best_fit})
-            best_fit = best_fit-up
-            if np.abs(np.max(up)) < 1e-4:
-                break
+        for i in range(self.maxit):
+            try:
+                up = self.tb.session.run(update, feed_dict={pars: best_fit})
+                best_fit = best_fit-self.relax*up
+                if np.abs(np.max(up)) < self.eps:
+                    break
+            except InvalidArgumentError:
+                o,p,g,h = self.tb.session.run([
+                    objective,
+                    pars,
+                    gradient,
+                    hessian,
+                ], feed_dict={best_fit: best_fit})
+                raise
```

Comment:
this will probably decrease coverage since we are adding lots of lines that we are not hitting in the tests. not sure how to cover it we would need to perhaps have a `validate_hessian` function that we could test stand-alone?